File: impl/app/android/app/src/main/java/com/ak/contexta/data/remote/dto/ChatRequest.kt
package com.ak.contexta.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class ChatCompletionRequest(
    val model: String,
    val messages: List<ChatMessage>,
    val temperature: Double = 0.7,
    val max_tokens: Int = 16384,
    val stream: Boolean = false
)

@Serializable
data class ChatMessage(
    val role: String, // "system" | "user" | "assistant"
    val content: String
)

File: impl/app/flutter/lib/data/remote/dto/chat_request.dart
/// DeepSeek 聊天补全请求 DTO。
/// 对照 Kotlin ChatRequest.kt（kotlinx.serialization，snake_case JSON 键）。
class ChatMessage {
  const ChatMessage({required this.role, required this.content});

  /// "system" | "user" | "assistant"
  final String role;
  final String content;

  Map<String, Object?> toJson() => {'role': role, 'content': content};
}

class ChatCompletionRequest {
  const ChatCompletionRequest({
    required this.model,
    required this.messages,
    this.temperature = 0.7,
    this.maxTokens = 16384,
    this.stream = false,
  });

  final String model;
  final List<ChatMessage> messages;
  final double temperature;
  final int maxTokens;
  final bool stream;

  Map<String, Object?> toJson() => {
        'model': model,
        'messages': [for (final m in messages) m.toJson()],
        'temperature': temperature,
        'max_tokens': maxTokens,
        'stream': stream,
      };
}
